synapse/handlers/room.py

```diff
@@ -237,30 +274,7 @@ async def _upgrade_room(
         )
 
         logger.info("Creating new room %s to replace %s", new_room_id, old_room_id)
-
-        # we create and auth the tombstone event before properly creating the new
-        # room, to check our user has perms in the old room.
-        (
-            tombstone_event,
-            tombstone_context,
-        ) = await self.event_creation_handler.create_event(
-            requester,
-            {
-                "type": EventTypes.Tombstone,
-                "state_key": "",
-                "room_id": old_room_id,
-                "sender": user_id,
-                "content": {
-                    "body": "This room has been replaced",
-                    "replacement_room": new_room_id,
-                },
-            },
-        )
-        old_room_version = await self.store.get_room_version(old_room_id)
-        validate_event_for_room_version(old_room_version, tombstone_event)
-        await self._event_auth_handler.check_auth_rules_from_context(
-            old_room_version, tombstone_event, tombstone_context
-        )
+        tombstone_event.content["replacement_room"] = new_room_id
```

Comment:
This is not the right way to update an event. It won't update the event hash and signature.